docs/docs/firmware/can-traffic.md

```diff
@@ -1,38 +1,46 @@
----
-title: Calculating CAN traffic on a bus
----
 
+# Calculating CAN traffic
 
+In order to calculate the CAN load, we must declare a few variables and their importance within the CAN protocol.
 
-In order to calculate the CAN load, we must declare a few variables and their importance within the CANbus protocol. 
-
-1. Baud: (kbaud)
+## Baud (kbaud)
 
 The baud rate is the rate at which data is transfered in a communication channel (like CAN). For example, 9600 Baud rate is 9600 bits per second
 
-2. Frequency (Hz)
+## Frequency (Hz)
 
-Frequency in the CANbus refers to how many CAN frames (or messages) are transmitted per second. So the greater the frequency, the more bits transferred, increasing the bus load. 
+Frequency is the number of messages transmitted per second for each message type. Different messages can have different frequencies.
 
-3. Total Data Per Cycle
+This variation affects the total bits transferred and, in turn, the bus load. To calculate bits per second, consider each message’s frequency and whether their lengths differ.
 
-Total data per cycle includes all the bits transmitted in one CAN message. This includes not only the payload but overhead bits like start of frame, CRC, acknowledgement, control field etc. 
+## Total Data Per Cycle
 
+Total data per cycle includes all the bits transmitted in one CAN message. This includes the payload and overhead bits like the start of frame, CRC, acknowledgment, control field, etc.
 
+## The bus load is determined by
 
-## The bus load is determined by: 
+$$
+\text{Total Bits Per Second} = \sum_{i=1}^n \left(\text{Frequency}_i \times \text{Message Length}_i\right)
+$$
 
-![Bus Load Formula](https://latex.codecogs.com/png.latex?\text{Bus%20Load%20(\%)}%20=%20\frac{\text{Number%20of%20Bits%20per%20Frame}%20\times%20\text{Number%20of%20Frames%20per%20Second}}{\text{CAN%20Speed}}%20\times%20100)
+$$
+\text{Bus Load (%)} = \frac{\text{Total Bits per Second}}{\text{Baud Rate}} \times 100
+$$
 
-![Bus Load Formula](https://latex.codecogs.com/png.latex?\text{Bus%20Load%20(\%)}%20=%20\frac{\text{Total%20Bits%20per%20Second}}{\text{CAN%20Speed}}%20\times%20100)
+## Sample calculation
 
-## Sample calculation: 
+**Given**:
 
-Total data/cycle: 1524 (bits per cycle)
+Battery Status: Frequency = 100Hz, Length = 400 bits
```

Comment:
after addressing the comment above, you should replace the `length` spec with `data length = x bytes` (you can choose x from 0-8) then calculate the number of bits from that. same for the next message

It would be cool to see this all in a table, like

| Message | Data Length (bytes) | Frequency (Hz) |
|:--------:|:-------:|:---:|
| BatteryStatus | 8 | 100 |
| MotorControl | 5 | 50 |

https://www.codecademy.com/resources/docs/markdown/tables